CRM/ManageLetterheads/Setup/CreateNavigationItem.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/**
+ * Setup Class for creating the Letterhead navigation item.
+ */
+class CRM_ManageLetterheads_Setup_CreateNavigationItem {
+
+  /**
+   * Used for running the upgrades.
+   *
+   * @return bool
+   *   TRUE when the upgrader runs successfully.
+   */
+  public function apply() {
+    $this->createManageLetterheadNavigationItem();
+
+    return TRUE;
+  }
+
+  /**
+   * Creates the letterhead navigation item if it doesn't already exist.
+   *
+   * The navigation item is created under Communications.
+   */
+  private function createManageLetterheadNavigationItem() {
+    $communications = civicrm_api3('Navigation', 'getsingle', [
+      'name' => 'Communications',
+      'parent_id' => 'Administer',
+    ]);
+    $letterheads = civicrm_api3('Navigation', 'get', [
+      'name' => 'Letterheads',
+    ]);
+
+    if (empty($letterheads['count'])) {
+      civicrm_api3('Navigation', 'create', [
+        'parent_id' => $communications['id'],
```

Comment:
Shouldnt we check for existence of communication like is it something that will always be there?

Reply:
Communications is part of the admin menu and it's probably something users won't be deleting. But if they do, that item is being requested using the `getsingle` action. It will break the installation from happening by throwing an error.

Should we handle this in a different way?